### DISCUSSION TITLE
parse concatenation columns in `relationship` method for `BelongsToSelect` field

### DISCUSSION THREAD

**Discussion by maherelgamil** [filamentphp/filament#1606] [Help] [answered]
Hello, 
Is there way to parse concatenation field as `displayname` attribute in r`elationship` as example:

```php
->relationship('instructor', 'first_name')
```

I need it to be something like this

```php
->relationship('instructor', \DB::raw("CONCAT(users.first_name, ' ', users.last_name) AS name"))
```

Thanks

**Comment by ryangjchandler** [comment, marked as answer]
@maherelgamil You could use a virtual column in the users table to create a `full_name` column. Although I do wonder if allowing `DB::raw` to be used in that place would be helpful.

**Comment by maherelgamil** [comment]
@ryangjchandler 
I think `virtual column` is a better solution for me

Thanks 🙏

**Comment by cornel-b** [comment]
Now it can be done as explained here https://filamentphp.com/docs/2.x/forms/fields#populating-automatically-from-a-relationship, using `getOptionLabelFromRecordUsing`
